src/Libraries/Microsoft.Extensions.Diagnostics.ResourceMonitoring/Linux/LinuxUtilizationParserCgroupV2.cs

```diff
@@ -86,47 +90,76 @@ internal sealed class LinuxUtilizationParserCgroupV2 : ILinuxUtilizationParser
     /// </summary>
     private static readonly FileInfo _cpuPodWeight = new("/sys/fs/cgroup/cpu.weight");
 
+    private static readonly FileInfo _cpuCgroupInfoFile = new("/proc/self/cgroup");
+
     private readonly IFileSystem _fileSystem;
     private readonly long _userHz;
 
+    // Cache for the trimmed path string to avoid repeated file reads and processing
+    private string? _cachedCgroupPath;
+
     public LinuxUtilizationParserCgroupV2(IFileSystem fileSystem, IUserHz userHz)
     {
         _fileSystem = fileSystem;
         _userHz = userHz.Value;
     }
 
-    public long GetCgroupCpuUsageInNanoseconds()
+    public string GetCgroupPath(string filename)
     {
-        // The value we are interested in starts with this. We just want to make sure it is true.
-        const string Usage_usec = "usage_usec";
-
-        // If the file doesn't exist, we assume that the system is a Host and we read the CPU usage from /proc/stat.
-        if (!_fileSystem.Exists(_cpuacctUsage))
+        // If we've already parsed the path, use the cached value
+        if (_cachedCgroupPath != null)
         {
-            return GetHostCpuUsageInNanoseconds();
+            return $"{PathPrefix}{_cachedCgroupPath}{filename}";
         }
 
         using ReturnableBufferWriter<char> bufferWriter = new(_sharedBufferWriterPool);
-        _fileSystem.ReadAll(_cpuacctUsage, bufferWriter.Buffer);
-        ReadOnlySpan<char> usage = bufferWriter.Buffer.WrittenSpan;
 
-        if (!usage.StartsWith(Usage_usec))
+        // Read the content of the file
+        _fileSystem.ReadFirstLine(_cpuCgroupInfoFile, bufferWriter.Buffer);
+        ReadOnlySpan<char> fileContent = bufferWriter.Buffer.WrittenSpan;
+
+        // Ensure the file content is not empty
+        if (fileContent.IsEmpty)
         {
-            Throw.InvalidOperationException($"Could not parse '{_cpuacctUsage}'. We expected first line of the file to start with '{Usage_usec}' but it was '{new string(usage)}' instead.");
+            Throw.InvalidOperationException($"The file '{_cpuCgroupInfoFile}' is empty or could not be read.");
         }
 
-        ReadOnlySpan<char> cpuUsage = usage.Slice(Usage_usec.Length, usage.Length - Usage_usec.Length);
+        // Find the index of the first colon (:)
+        int colonIndex = fileContent.LastIndexOf(':');
+        if (colonIndex == -1 || colonIndex + 1 >= fileContent.Length)
+        {
+            Throw.InvalidOperationException($"Invalid format in file '{_cpuCgroupInfoFile}'. Expected content with ':' separator.");
+        }
 
-        int next = GetNextNumber(cpuUsage, out long microseconds);
+        // Extract the part after the last colon and cache it for future use
+        ReadOnlySpan<char> trimmedPath = fileContent.Slice(colonIndex + 1);
+        _cachedCgroupPath = trimmedPath.ToString().TrimEnd('/') + "/";
```

Comment:
can we do something like this, e.g. add PathPrefix immediately:
```suggestion
        _cachedCgroupPath = $"{PathPrefix}{trimmedPath.ToString().TrimEnd('/') + "/"}";
```
so that we don't have to add it in `return` statement on lines 112 and 138

Reply:
in this case we can just remove the unused field. In fact, we don't even need the field, we can just insert the string /sys/fs/cgroup directly in here